src/wagtail_analytics/lib/analytics.py

```diff
@@ -0,0 +1,223 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from datetime import date, datetime, timedelta
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
+
+import ipdb
```

Comment:
format your imports and code please, ipdb should not be here
also the sorting order is wrong